packages/kotti-ui/source/kotti-popover/KtPopover.vue

```diff
@@ -19,13 +26,27 @@
 import { createPopper } from '@popperjs/core'
 import { mixin as clickaway } from 'vue-clickaway'
 
+import KtPopoverIconTextItem from './KtPopoverIconTextItem.vue'
+
+const optionIsValid = (option) =>
+	(!option.isDisabled || typeof option.isDisabled === 'boolean') &&
+	option.onClick &&
+	typeof option.onClick === 'function' &&
+	option.label &&
```

Comment:
this check means you disregard an empty string label
your API shouldn't prevent the user from having an empty string for a label if they want to
so this check is unneeded

usually for primitives, you just check that you get the type exactly of that primitive

Reply:
@carsoli I think it’s fine to check for a missing label. Why would we want an option in this dropdown that doesn’t have any content? The user will be confused and I see no way that this would ever be anything but a mistake

Reply:
@carsoli clarified that this is due to also supporting icons, so we may want icon-only entries. Please disregard my comment.